src/interfaces/background_field_interface.jl

```diff
@@ -26,7 +26,7 @@ Abstract base type for pulsed plane-wave fields.
 
     reference_momentum(::AbstractPulsedPlaneWaveField)
     domain(::AbstractPulsedPlaneWaveField)
-    phase_duration(::AbstractPulsedPlaneWaveField)
+    pulse_duration(::AbstractPulsedPlaneWaveField)
```

Comment:
... while length and width indicate some distance in space. 'extent' is fine, but may also be confused with the result of the `domain()` function (which we could have named 'support', I guess). This is soo hard :rofl:

Reply:
This is why I thought `phase_duration` or `phase_width` is conceivable because these directly indicate, that they mean the extent in the *phase*, and not in any space-time domain. 

Speaking of `domain`, I like the idea of renaming it to `extent`, but I would like to stay in line with `DomainSets`, where this is called *domain* and not *support*. It is distinct from `phase_duration` because the latter is just a parameter, whereas the former is an interval, which can be used for integrals like Fourier transforms and such.

Reply:
I renamed to `pulse_length` now, since:
* you seem to be fine with it
* it does express something longitudinal and not transverse (in opposition to 'width')
* it is clear that this is a typical measure for the extent but not the *full* extent due its its common usage
* it is clear that this quantity relates to the full pulse and not just one field oscillation (in opposition to 'phase')

If @szabo137 is fine with this, I will squash and then it can be merged, no?

Reply:
I will this conversation unresolved in order to keep it visible.